cmd/difference.go

```diff
@@ -335,12 +336,20 @@ func difference(ctx context.Context, sourceClnt, targetClnt Client, isMetadata b
 		err := differenceInternal(ctx, sourceClnt, targetClnt, isMetadata, isRecursive, returnSimilar, dirOpt, diffCh)
 		if err != nil {
 			// handle this specifically for filesystem related errors.
-			switch err.ToGoError().(type) {
+			switch v := err.ToGoError().(type) {
 			case PathNotFound, PathInsufficientPermission:
 				diffCh <- diffMessage{
 					Error: err,
 				}
 				return
+			case minio.ErrorResponse:
+				switch v.Code {
+				case "NoSuchBucket", "NoSuchKey":
```

Comment:
differenceInternal is doing listing - I don't think we will ever have 'NoSuchKey' here